Makefile

```
@@ -1,16 +1,28 @@
 #############################################################################
 # Makefile for building: TrueNAS SCALE
 #############################################################################
+PYTHON?=/usr/bin/python3
+COMMIT_HASH=$(shell git rev-parse --short HEAD)
+
+check:
+ifeq ("$(wildcard ./venv-${COMMIT_HASH})","")
+	@rm -rf venv-*
+	@${PYTHON} -m pip install -U virtualenv >/dev/null 2>&1
+	@${PYTHON} -m venv venv-${COMMIT_HASH}
+	@. ./venv-${COMMIT_HASH}/bin/activate && \
+		python3 -m pip install -r requirements.txt >/dev/null 2>&1 && \
```

Will this fail silently in case of an error?